File: app/src/main/java/com/depoza/util/timeperiod/TimeProvider.java
package com.depoza.util.timeperiod;

/**
 * Provider of the current time in milliseconds
 */
@FunctionalInterface
public interface TimeProvider {
    long currentTimeMls();

    /**
     * {@link TimeProvider} which uses {@link System#currentTimeMillis()}
     */
    class DefaultTimeProvider implements TimeProvider {

        @Override
        public long currentTimeMls() {
            return System.currentTimeMillis();
        }
    }
}

File: app/src/main/java/com/depoza/util/timeperiod/TimeProvider.kt
package com.depoza.util.timeperiod

/**
 * Provider of the current time in milliseconds
 */
@FunctionalInterface
interface TimeProvider {
    fun currentTimeMls(): Long

    /**
     * [TimeProvider] which uses [System.currentTimeMillis]
     */
    class DefaultTimeProvider : TimeProvider {

        override fun currentTimeMls(): Long {
            return System.currentTimeMillis()
        }
    }
}
